Apps/JobScheduler/src/Jobs/NotifyUpdatedLegalAgreementsJob.cs

```diff
@@ -73,29 +87,63 @@ public class NotifyUpdatedLegalAgreementsJob
+                DBResult<LegalAgreement> legalAgreementsResult = this.legalAgreementDelegate.GetActiveByAgreementType(lac.Code);
+                if (legalAgreementsResult.Status == DBStatusCode.Read)
                 {
-                    this.logger.LogInformation($"Fetching {document.LastCheckedKey} from application settings");
-                    ApplicationSetting lastCheckedSetting = this.applicationSettingsDelegate.GetApplicationSetting(ApplicationType.JobScheduler, this.GetType().Name, document.LastCheckedKey);
-                    this.logger.LogInformation($"Found {document.LastCheckedKey} with value of {lastCheckedSetting.Value}");
-                    DateTime lastChecked = System.DateTime.Parse(lastCheckedSetting.Value!, CultureInfo.InvariantCulture);
-                    if (result.Payload.EffectiveDate > lastChecked)
-                    {
-                        // TODO: Pull All profiles
-                        // TODO: For each profile send email template
-                    }
+                    this.ProcessLegalAgreement(legalAgreementsResult.Payload, lac);
                 }
                 else
                 {
-                    this.logger.LogCritical($"Unable to read {document.Name} from the DB ABORTING...");
+                    this.logger.LogCritical($"Unable to read {lac.Name} from the DB ABORTING...");
                 }
             }
         }
+
+        private void ProcessLegalAgreement(LegalAgreement agreement, LegalAgreementConfig config)
+        {
+            this.logger.LogInformation($"{config.Name} found, last updated {agreement.EffectiveDate}");
+            this.logger.LogInformation($"Fetching {config.LastCheckedKey} from application settings");
+            ApplicationSetting lastCheckedSetting = this.applicationSettingsDelegate.GetApplicationSetting(ApplicationType.JobScheduler, this.GetType().Name, config.LastCheckedKey);
+            this.logger.LogInformation($"Found {config.LastCheckedKey} with value of {lastCheckedSetting.Value}");
+            DateTime lastChecked = System.DateTime.Parse(lastCheckedSetting.Value!, CultureInfo.InvariantCulture);
+            if (agreement.EffectiveDate > lastChecked)
+            {
+                Dictionary<string, string> keyValues = new Dictionary<string, string>();
+                keyValues.Add("host", this.host);
+                keyValues.Add("Path", config.Path);
+                keyValues.Add("EffectiveDate", agreement.EffectiveDate.Value.ToString("MMMM dd, yyyy", CultureInfo.InvariantCulture));
+                keyValues.Add("ContactEmail", config.ContactEmail);
+                int page = 0;
+                DBResult<List<UserProfile>> profileResult;
+                do
+                {
+                    profileResult = this.profileDelegate.GetAllUserProfilesAfter(lastChecked, page, this.profilesPageSize);
+                    foreach (UserProfile profile in profileResult.Payload)
```

Comment:
instead of calling the service multiple times with a single email entry, how about calling it once with a list of profile.Emails. That way the service does not need to hold the transaction, only the delegate

Reply:
As discussed, this gives control of the transaction to the email service when it needs to stay with the batch program.